constraint_utils/constraint_basics.py

```diff
@@ -843,12 +925,43 @@ def oe_confidence_interval(ht: hl.Table, obs: hl.expr.Int32Expression, exp: hl.e
 
 
 def calculate_z(input_ht: hl.Table, obs: hl.expr.NumericExpression, exp: hl.expr.NumericExpression, output: str = 'z_raw') -> hl.Table:
+    """
+    Compute the signed raw z score using observed and expected variant counts.
+    
+    The raw z scores are positive when the transcript had fewer variants than expected, and are negative when transcripts had more variants than expected.
+    
+    Function will have following annotations in the output Table in addition to keys:
+        - `output` - the raw z score
+
+    :param input_ht: Input Table.
+    :param obs: Observed variant counts.
+    :param exp: Expected variant counts.
+    :param output: The column name for raw z score, defaults to 'z_raw'.
+    :return: A Table with raw z score.
+    """
     ht = input_ht.select(_obs=obs, _exp=exp)
     ht = ht.annotate(_chisq=(ht._obs - ht._exp) ** 2 / ht._exp)
     return ht.select(**{output: hl.sqrt(ht._chisq) * hl.cond(ht._obs > ht._exp, -1, 1)})
 
 
 def calculate_all_z_scores(ht: hl.Table) -> hl.Table:
+    """
+    Calculate z scores for synomynous variants, missense variants, and LoF variants.
+    
+    z score = {variant_type}_z_raw / {variant_type}_sd (variant_type could be syn, mis, or lof)
+    
+    Function will add the following annotations to output Table:
+        - syn_sd (global) - standard deviation of synonymous variants raw z score
+        - mis_sd (global) - standard deviation of missense varinats raw z score
+        - lof_sd (global) - standard deviation of LoF variants raw z score
+        - constraint_flag - flags to determine if the variant is 'no_variants', 'no_exp_syn', 'no_exp_mis', 'no_exp_lof', 'syn_outlier', 'mis_too_many', or 'lof_too_many'
```

Comment:
I would add what each of these flags means

From supplement:

constraint_flag: Reason gene does not have constraint metrics. One of:
no variants: Zero observed synonymous, missense, pLoF variants
no_exp_syn: Zero expected synonymous variants
no_exp_mis: Zero expected missense variants
no_exp_lof: Zero expected pLoF variants
syn_outlier: Too many or too few synonymous variants; synonymous z score < -5 or synonymous z score > 5
mis_too_many: Too many missense variants; missense z score < -5
lof_too_many: Too many pLoF variants; pLoF z score < -5